### PR TITLE
libgtkflow: init at 0.8.0

### PR DESCRIPTION
**Pull request by grindhold**
###### Description of changes

Packaged libgtkflow

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libgtkflow/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  mesonFlags = [
+    "-Denable_valadoc=true"
+  ];
+
+  meta = with lib; {
+    description = "Flow graph widget for Gtk3";
+    homepage = "https://notabug.org/grindhold/libgtkflow";
+    maintainers = with maintainers; [ grindhold ];
+    license = licenses.lgpl3;
```
Comment (jtojnar) [inline pkgs/development/libraries/libgtkflow/default.nix:34]: `lgpl3` is unclear, use `lgpl3Plus` or `lgpl3Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.

File: pkgs/development/libraries/libgtkflow/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    gtk3
+    glib
+  ];
+
+  mesonFlags = [
+    "-Denable_valadoc=true"
+  ];
+
+  meta = with lib; {
+    description = "Flow graph widget for Gtk3";
```
Comment (jtojnar) [inline pkgs/development/libraries/libgtkflow/default.nix:31]:
```suggestion
    description = "Flow graph widget for GTK 3";
```

**Review by jtojnar — COMMENTED** [review]
Thanks, this was on my to-do list.

File: pkgs/development/libraries/libgtkflow/default.nix (hunk 1 above)
Comment (grindhold) [reply pkgs/development/libraries/libgtkflow/default.nix]: done

File: pkgs/development/libraries/libgtkflow/default.nix (hunk 2 above)
Comment (grindhold) [reply pkgs/development/libraries/libgtkflow/default.nix]: done

Comment (jtojnar) [reply pkgs/development/libraries/libgtkflow/default.nix]: The GTK name should be capitalized.

Comment (grindhold) [reply pkgs/development/libraries/libgtkflow/default.nix]: done

**Review by jtojnar — APPROVED** [review]

File: pkgs/development/libraries/libgtkflow/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{stdenv, lib, vala, meson, ninja, pkg-config, fetchgit, gobject-introspection, glib, gtk3}:
+
+stdenv.mkDerivation rec {
+  pname = "libgtkflow";
+  version = "0.8.0";
+
+  src = fetchgit {
+    url = "https://notabug.org/grindhold/libgtkflow";
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libgtkflow/default.nix:9]:
```suggestion
    rev = version;
```

File: pkgs/development/libraries/libgtkflow/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
+{stdenv, lib, vala, meson, ninja, pkg-config, fetchgit, gobject-introspection, glib, gtk3}:
+
+stdenv.mkDerivation rec {
+  pname = "libgtkflow";
+  version = "0.8.0";
+
+  src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libgtkflow/default.nix:7]: Please use fetchGitea.

File: pkgs/development/libraries/libgtkflow/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  meta = with lib; {
+    description = "Flow graph widget for GTK 3";
+    homepage = "https://notabug.org/grindhold/libgtkflow";
+    maintainers = with maintainers; [ grindhold ];
+    license = licenses.lgpl3Plus;
+    platforms = platforms.all;
+  };
+}
+
+
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libgtkflow/default.nix:39]:
```suggestion
    platforms = platforms.unix;
  };
}
```

File: pkgs/development/libraries/libgtkflow/default.nix [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  meta = with lib; {
+    description = "Flow graph widget for GTK 3";
+    homepage = "https://notabug.org/grindhold/libgtkflow";
+    maintainers = with maintainers; [ grindhold ];
+    license = licenses.lgpl3Plus;
+    platforms = platforms.unix;
+  };
+}
+
+
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libgtkflow/default.nix:41]:
```suggestion
}
```

Comment (grindhold) [reply pkgs/development/libraries/libgtkflow/default.nix]: done

**Pull request merged**